Fix a bug which caused wdfs to freeze when unmounting if it wasn't called in
"foreground" mode.

Index: src/wdfs-main.c
--- src/wdfs-main.c.orig
+++ src/wdfs-main.c
@@ -27,6 +27,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stdint.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <assert.h>
@@ -1219,6 +1220,11 @@ static int wdfs_statfs(const char *localpath, struct s
 	static void* wdfs_init()
 #endif
 {
+	/* We initialize the cache here 
+	because otherwise the cache thread
+	wont survive fork() if wdfs is called
+	as a daemon */
+	cache_initialize();
 	if (wdfs.debug == true)
 		fprintf(stderr, ">> %s()\n", __func__);
 	return NULL;
@@ -1387,7 +1393,6 @@ int main(int argc, char *argv[])
 		}
 	}
 
-	cache_initialize();
 
 	/* finally call fuse */
 	status_program_exec = call_fuse_main(&options);
